tlatools/org.lamport.tlatools/src/tla2sany/drivers/SemanticException.java

```diff
@@ -1,4 +1,56 @@
 // Copyright (c) 2003 Compaq Corporation.  All rights reserved.
 package tla2sany.drivers;
 
-public class SemanticException extends Exception {}
+import tla2sany.semantic.AbortException;
+import tla2sany.semantic.Errors;
+import tla2sany.semantic.Errors.ErrorDetails;
+
+/**
+ * User-facing exception class for errors occurring during later stages
+ * of the parse process. Translates internal-only {@link AbortException}
+ * instances into external-ready {@link SemanticException} instances. For
+ * errors occurring earlier during the syntax processing phase of the parse
+ * process, consider using {@link tla2sany.parser.ParseException} instead.
+ */
+public class SemanticException extends Exception {
+
+  /**
+   * The internal error wrapped by this exception.
+   */
+  private final AbortException internalException;
+
+  /**
+   * Initializes a new instance of the {@link SemanticException} class.
+   *
+   * @param internalException The internal exception causing the error.
+   */
+  public SemanticException(AbortException internalException) {
```

Comment:
This constructor should either:
 1. take `ErrorDetails` and `Errors` instead of an `AbortException`, or
 2. call `super(internalException)` to mark the abort as this exception's cause.

My preference is for the second, but I'd be happy with either.

Java exceptions are chained: each exception can point to a [cause](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Throwable.html#getCause()), so an exception can carry complete information about both the original error and which components caught and re-threw it.

By convention, an exception constructor that takes an exception as its argument is meant to build a link in the cause chain. It's a little weird that this class doesn't do that.

Reply:
Seems good to me!